lib/cucumber/core/test/runner.rb

```diff
@@ -4,40 +4,39 @@ module Cucumber
   module Core
     module Test
       class Runner
-        attr_reader :report, :running_test_case, :running_test_step
-        private :report, :running_test_case, :running_test_step
+        attr_reader :events, :running_test_case, :running_test_step
+        private :events, :running_test_case, :running_test_step
 
-        def initialize(report)
-          @report = report
+        def initialize(events)
+          @events = events
         end
 
         def test_case(test_case, &descend)
           @running_test_case = RunningTestCase.new
           @running_test_step = nil
-          report.before_test_case(test_case)
+          events.test_case_starting(test_case)
           descend.call(self)
```

Comment:
IIRC the idea was that by naming the proc (rather than using the implicit anonymous block) we made it a bit clearer what that proc was going to do. TBH we could probably just iterate over `test_case.test_steps` and not worry about using the double-dispatch malarkey. At the time @tooky and I wrote this I'd just read [Object Thinking](http://www.goodreads.com/book/show/43940.Object_Thinking) and was on a deep trip into tell-don't-ask.